### PR TITLE
NK-483 NK-549

### PR DESCRIPTION
**Pull request by guymenahem**
NK-483 NK-549

### CODE REVIEW

**Review by bazulay — APPROVED** [review]

File: inaugurator/tests/test_ipmi.py [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+        self.mock_sh.setCommands(cmds)
+        ipmi_some.configureIPMI()
+
+    def test_kernelConfigureIPMIDoNothing(self):
+        """
+        test that each subset of values is not doing any configuration, as expected
+        @return: None
+        @rtype: None
+        """
+        # username and password only - do nothing
```
Comment (roeigo-stratoscale) [inline inaugurator/tests/test_ipmi.py:113]: user or password

**Review by roeigo-stratoscale — APPROVED** [review]

**Review by roeigo-stratoscale — CHANGES_REQUESTED** [review]
My previous review was approved, by mistake.
Please see ,y comments before merging

**Review by Slevy35 — APPROVED** [review]

File: inaugurator/IPMIDriver.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    def _verifyModuleLoaded(self, module):
+        """
+        verify kernel module is loaded. if not - raises exception.
+        @param module: module name
+        @type module: string
+        @return: None - raise exception if kernel module not loaded
+        """
+        output = sh.run("lsmod | awk '{print $1}' | grep %s | wc -l" % module)
+
+        if int(output) < 1:
```
[the remark this answers is not in the document]
Comment (guymenahem) [reply inaugurator/IPMIDriver.py:44]: No, If the specific module is loaded I would expect to get 1 and if not 0

File: inaugurator/IPMIDriver.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            raise Exception("Cannot find kernel module for IPMI- module %s. all loaded IPMI modules are %s."
```
[the remark this answers is not in the document]
Comment (guymenahem) [reply inaugurator/IPMIDriver.py:45]: You may weren't able to load any module but you may load 2 and the last wasn't load.
The kernel module ipmi_si is loaded only if you the BMC found.

If none of the modules is loaded you will get an empty output there.

File: inaugurator/tests/test_ipmi.py (hunk 1 above)
Comment (guymenahem) [reply inaugurator/tests/test_ipmi.py]: fixed

**Pull request merged**